### DISCUSSION TITLE
Automatic Windows installation with Cobbler

### DISCUSSION THREAD

**Discussion by badacook** [cobbler/cobbler#3318] [General]
<big><font face="楷体">**Hi cobbler guys** </font></big><br>
&#8195; Follow the  topic 'Automatic Windows installation with Cobbler' of cobbler  document, as a Maintainer point cobbler added Windows Support until v3.2.1 officially,<br> And i had attempted to deploy Windows and Linux on cobbler server but failed,so  **I want to know the officiall OS test environment of cobbler for Windows deployment,<br>such  as the os distribution、version, the package source and some key related package version details**.Any help will be great,thanks.

**Comment by SchoolGuy** [comment]
@tpw56j Maybe you can chime in here? I havn't done a hands-on test, so you are the only one afaik who used it successfully.

**Reply by tpw56j** [reply]
@SchoolGuy Okay, I'll try to answer the questions.

**Comment by tpw56j** [comment]
@badacook I tested installing Windows on different versions of Fedora, but I think any distribution that has packages will do: `python3-pefile` `python3-hivex` `wimlib-utils`.
Windows related documentation is here: https://github.com/cobbler/cobbler/blob/main/docs/user-guide/wingen.rst.
If something does not work, then tell me the version of your distribution and I will try to reproduce the installation.

**Reply by badacook** [reply]
thanks for your reply ,i have install cobbler  on Fedora Server version  35 ,but i  have met two problem below 
1. When all services  like httpd cobblerd rsyncd dhcpd tftp are running ,i import Fedora Server version  35 iOS without error ,but When boot from network on  the test client ,it get IP from DHCP ,but the "PEX-T01：file not found " error  occur ,but i have install syslinux,and the dhcp.template have the filename secction config 
2. from cobbler signature report, we know the signature distribution os, is there a way to signature for another distribution like Centos 
Any help will be great,thanks.

**Reply by tpw56j** [reply]
1. The difference in preparing the environment for installing Windows from Linux is the need to install 3 additional packages, configure `samba` and `/etc/tftpd.rules`. First, it's a good idea to make sure your environment can install Linux, and then you can start configuring your Windows installation.
Check the contents of `/etc/dhcp/dhcpd.conf` and that all "filenames" exist in the tftp directory.
For example:
```
# cat /etc/dhcp/dhcpd.conf
   ..
     class "pxeclients" {
        match if substring (option vendor-class-identifier, 0, 9) = "PXEClient";

        if exists user-class and option user-class = "iPXE" {
            filename "/ipxe/default.ipxe";
        }
        else if option arch = 00:07 {
            filename "shim.efi";
        }
        else
        {
            filename "pxelinux.0";
        }

        next-server 192.168.130.1;
     }
   ..
# ls -l /var/lib/tftpboot/{ipxe/default.ipxe,shim.efi,pxelinux.0}
-rw-r--r--. 1 root root   44517 Nov 25 20:40 /var/lib/tftpboot/ipxe/default.ipxe
-rw-r--r--. 1 root root   16794 Jul  6  2020 /var/lib/tftpboot/pxelinux.0
-rwxr-xr-x. 1 root root 1210776 Oct  2  2018 /var/lib/tftpboot/shim.efi
```
2. The file https://github.com/cobbler/cobbler/blob/main/config/cobbler/distro_signatures.json contains signatures for many OS versions, including CentOS and Windows.

In order for me to help prepare the installation environment, please provide more details - snippets of tftp logs, dhcpd logs, etc.

**Comment by badacook** [comment]
thanks for you nice help, as this article https://github.com/cobbler/cobbler/issues/2844?ysclid=lcpm5lvxrp548089307 point Linux bootLoader missing can solve by running /usr/share/cobbler/bin/mkgrub.sh ,
And i have been properly tested Fedora 35 automatic installation with Cobbler , the next step for Windows dependent , but what make me confused is the official document guiding significance,
such as the config windows_enabled: true in /etc/cobbler/settings.yaml is  not supported ,but is supported with  Cobbler 3.3.3 on Fedora 36,but the bootLoader missing problem need another  way to solve ,
beyond  that the signatures problem remains , simply  replace the distro_signatures.json would come across irreversible error , as the cobbler processing combine closely connection and cobbler Version iteration,
what makes me tired ,and i just want to perform the wide range os deployment  as possible on a certain distribution and cobbler version with step by step , but follow the document always failed  and cost a lot of time,
Now i just want to know the  official document is suitable for what environment and the environment details ,Any help will be great,thanks.

**Reply by tpw56j** [reply]
Each version of Fedora supports a specific version of Cobbler.  The maintainers of the distribution packages ensure that the versions of the packages are consistent with each other.
For example, Fedora 36 officially supports `cobbler-3.3.3-1.fc36.noarch.rpm`, and its Windows settings are located in the `/etc/cobbler/settings.d/windows.settings` file. This package also contains the official documentation for version 3.3.3 of Cobbler.
With each new version, something changes, the history of changes can be viewed at https://github.com/cobbler/cobbler/releases.

**Comment by badacook** [comment]
First of all thanks for your help , with your advise I deploy cobbler version 3.3.3 on Fedora Server 37 to avoid difference between os distribution and cobbler version .
And fellow the official document to test Linux and Windows deployment , It is succeed to deploy Fedora 35 client 、 Centos 7，but Windows 10 Without the boot menu,I donot know
which step cause this problem,i had posted a discussion [Windows 10 PXE boot deployment without boot menu](https://github.com/cobbler/cobbler/discussions/3327),thought that you are experience with cobbler,
I am looking forward to your advice for your convenient ,Any help will be great thanks.
